File: Solution/0206.Reverse_Linked_List/0206.Reverse_Linked_List.cpp
//迭代法
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode *pre=nullptr,*node=head;
        while(node)
        {
            ListNode *s=node->next;
            node->next=pre;
            pre=node;
            node=s;
        }
        return pre;
    }
};
//递归法
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if(!head||!head->next)
            return head;
        ListNode *newhead=reverseList(head->next);
        head->next->next=head;
        head->next=nullptr;
        return newhead;

    }
};

File: Solution/0206.Reverse_Linked_List/0206.Reverse_Linked_List.py
#迭代法
class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        pre,node=None,head
        while node:
            s=node.next
            node.next=pre
            pre,node=node,s
        return pre
#递归法
class Solution:
    def reverseList(self, head: ListNode) -> ListNode:
        if head is None or head.next is None:
            return head
        newhead=self.reverseList(head.next)
        head.next.next=head
        head.next=None
        return newhead
